src/gui/qgsexpressionpreviewwidget.cpp

```diff
@@ -55,10 +55,12 @@ void QgsExpressionPreviewWidget::setCurrentFeature( const QgsFeature &feature )
     QgsFeature validFeature( feature );
     validFeature.setValid( true );
     mExpressionContext.setFeature( validFeature );
+    mFeaturePickerWidget->setEnabled( false );
```

Comment:
```suggestion
    mFeaturePickerWidget->setTooltip( tr( "No feature was found on this layer to test the expression." ) );
```

Add that and we've got a good reason for not hiding the widget ;)